src/mixins/object_geometry.mixin.js

```diff
@@ -88,10 +88,29 @@
      * The coords are returned in an array.
      * @return {Array} [tl, tr, br, bl] of points
      */
-    getCoords: function(absolute, calculate) {
+    getCoords: function (absolute, calculate) {
       return arrayFromCoords(this._getCoords(absolute, calculate));
     },
 
+    /**
+     * @returns {{ offset: fabric.Point, blur: fabric.Point }}
+     */
+    getShadowOffsets: function () {
+      var blur = this.shadow.blur;
+      var shadowOffset = new fabric.Point(this.shadow.offsetX, this.shadow.offsetY),
+          blurOffset = new fabric.Point(blur, blur);
+      if (!this.shadow.nonScaling) {
+        var scaling = this.getObjectScaling(),
+            sx = scaling.scaleX, sy = scaling.scaleY;
+        shadowOffset.setXY(shadowOffset.x * sx, shadowOffset.y * sy);
+        blurOffset.setXY(blurOffset.x * sx, blurOffset.y * sy);
```

Comment:
instead of creating the point and then set their values again, i would work on the scale factor, start at zoom level ( or 1 ) and then grows with sx and sy, and you create the points at the end.

Reply:
done